sycl/include/sycl/sycl.hpp

```diff
@@ -8,6 +8,8 @@
 
 #pragma once
 
+#include <sycl/core.hpp>
+
 #include <sycl/accessor.hpp>
```

Comment:
I think we should remove includes coming from core.hpp.

Reply:
I'm not so sure we should do it now. I imagine next few steps would be related to cleaning some inter-header dependencies (like 70% of the `queue.hpp`), and we will be able to do the cleanups at that time. Another short-term goal is to switch even more tests to the `<sycl/core.hpp>` so that any optimizations could be actually seen in the CI. In this PR I've only updated the tests that were trivial for me to update.

Reply:
Updated as requested.